### DISCUSSION TITLE
oidc_jwk_list_destroy_hash segfault since v2.4.6

### DISCUSSION THREAD

**Discussion by bommo1** [OpenIDC/mod_auth_openidc#990] [General]
until 2.4.5 it works. 2.4.6 - 2.4.12.2 segfault

oidc_jwk_list_destroy_hash -> apr_hash_first -> apr_hash_next -> segfault on line `hi->this = hi->ht->array[hi->index++];`

Apache 2.4.55.

Config:
```
OIDCOAuthAcceptTokenAs header
OIDCOAuthVerifyCertFiles kid#/file.pem
OIDCAuthNHeader X-Forwarded-User
OIDCOAuthRemoteUserClaim preferred_username
```

Probably from this commit: https://github.com/zmartzone/mod_auth_openidc/commit/2019f5421103a2114bb544a5ce01ac074842aac4

**Comment by zandbelt** [comment]
I think it was just fixed in https://github.com/zmartzone/mod_auth_openidc/commit/203b764946cfa194bd8365d8cbb99ed68e0cd542 can you confirm?

**Comment by bommo1** [comment]
Thank you for the quick response!
We have 2 VirtualHosts each with only one cert in OIDCOAuthVerifyCertFiles.
I cannot quickly build from source at this point. I'll test and confirm when the new version is released.

**Reply by zandbelt** [reply]
for which platform you'd need a build to quickly test?

**Comment by bommo1** [comment]
AlmaLinux 8.6  x86_64

**Reply by zandbelt** [reply]
please try https://www.mod-auth-openidc.org/downloads/mod_auth_openidc-2.4.12.3rc0-1.el8.x86_64.rpm

**Reply by bommo1** [reply]
I confirm that the problem is fixed in this build. Thank you!

**Comment by zandbelt** [comment]
now published as part of release 2.4.12.3, see: https://github.com/zmartzone/mod_auth_openidc/releases/tag/2.4.12.3
